tsl/src/compression/compression.h

```diff
@@ -248,6 +248,8 @@ typedef struct RowCompressor
 	int64 num_compressed_rows;
 	/* if recompressing segmentwise, we must know this so we can reset the sequence number */
 	bool segmentwise_recompress;
+	/* flag for checking if we are working on the first tuple */
+	bool first_iteration;
```

Comment:
this `bool` would not be needed in case `row_compressor_new_row_is_in_new_group()` could answer true on the first call

I had some not-so-nice ideas like:
* init segment_info with `is_null` but with `1` value
* `row_compressor->rows_compressed_into_current_value` is only `0` during the first invocation; after that we will always have an open row with `>=1` entries

I think they are not necessarily better - I'll just leave them here for a second consideration by you...

Reply:
Had some similar ideas but decided to go with this explicit way for a couple of reasons:
- matches the previous code more closely
- it makes a bit more understandable without possibility to introduce subtle bugs by relying on other fields